compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2842,6 +2854,10 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
       def Synthetic: Flags = dotc.core.Flags.Synthetic
       def Trait: Flags = dotc.core.Flags.Trait
       def Transparent: Flags = dotc.core.Flags.Transparent
+
+      private[QuotesImpl] def validMethodFlags: Flags = Private | Protected | Override | Deferred | Final | Method | Implicit | Given | Local | JavaStatic | AbsOverride // Synthetic | ExtensionMethod | Exported | Erased | Infix | Invisible
```

Comment:
- It looks like the list here is duplicated in the documentation of newMethod, we should add a comment here to remember to update the list in the documentation.
- I don't think users can create JavaStatic methods.